atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WriteOnceDeleteOnceWorkflows.java

```diff
@@ -0,0 +1,106 @@
+ */
+public final class WriteOnceDeleteOnceWorkflows {
+
+    private static final SecureRandom RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
+
+    private static final Integer COLUMN = 0;
+
+    private WriteOnceDeleteOnceWorkflows() {
+        // Utility class
+    }
+
+    public static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor) {
+        return create(store, configuration, executionExecutor, DefaultBuggifyFactory.INSTANCE, RANDOM::nextInt);
+    }
+
+    @VisibleForTesting
+    static Workflow create(
+            InteractiveTransactionStore store,
+            WriteOnceDeleteOnceWorkflowConfiguration configuration,
+            ListeningExecutorService executionExecutor,
+            BuggifyFactory buggifyFactory,
+            IntFunction<Integer> randomNumberGenerator) {
+        WriteOnceDeleteOnceWorkflow workflow =
+                new WriteOnceDeleteOnceWorkflow(buggifyFactory, randomNumberGenerator, configuration);
+        return DefaultWorkflow.create(store, workflow::run, configuration, executionExecutor);
+    }
+
+    private static class WriteOnceDeleteOnceWorkflow {
+        private final BuggifyFactory buggifyFactory;
+        private final IntFunction<Integer> rowKeyGenerator;
+        private final WriteOnceDeleteOnceWorkflowConfiguration configuration;
+
+        public WriteOnceDeleteOnceWorkflow(
+                BuggifyFactory buggifyFactory,
+                IntFunction<Integer> rowKeyGenerator,
+                WriteOnceDeleteOnceWorkflowConfiguration configuration) {
+            this.buggifyFactory = buggifyFactory;
+            this.rowKeyGenerator = rowKeyGenerator;
+            this.configuration = configuration;
+        }
+
+        public Optional<WitnessedTransaction> run(InteractiveTransactionStore store, int taskIndex) {
+            configuration.transactionRateLimiter().acquire();
+            return store.readWrite(txn -> {
+                Integer rowKey = rowKeyGenerator.apply(configuration.maxKey());
+                if (buggifyFactory
+                        .maybe(calculateChanceForDelete(taskIndex, configuration.iterationCount()))
+                        .asBoolean()) {
+                    txn.delete(configuration.tableConfiguration().tableName(), cell(rowKey));
+                } else {
+                    txn.write(configuration.tableConfiguration().tableName(), cell(rowKey), RANDOM.nextInt(10000));
+                }
+            });
+        }
+    }
+
+    @VisibleForTesting
+    static Double calculateChanceForDelete(int taskIndex, int iterationCount) {
+        // Index generation is [inclusive, exclusive], so we need to add 1 to the task index.
+        return (taskIndex + 1.0) / iterationCount;
+    }
+
+    @VisibleForTesting
+    static WorkloadCell cell(Integer index) {
```

Comment:
nit: we've used boxed and unboxed int across the methods, we should probably pick a convention and stick with it!